pkg/kubelet/kubelet.go

```diff
@@ -2665,49 +2633,28 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 			// We failed pods that we rejected, so allocatedPods include all admitted
 			// pods that are alive.
 			allocatedPods := kl.getAllocatedPods()
```

Comment:
I just realized - I think there's a race condition here? If a resize happens between here and calling `allocationManager.AddPod`.

Allocation manager is already the source of truth for the allocations though, so I think the fix should be to simply pass the active pods to `allocationManager.AddPod`, and handle the conversion to the allocated resources within the allocation manager while holding the lock.

Reply:
fixed, thanks for the catch